enterprise/frontend/src/embedding/auth-common/validate-session-token.ts

```diff
@@ -26,10 +26,24 @@ export function validateSessionToken(session: any) {
 
   // We should also receive `iat` and `status` in the response, but we don't actually need them
   // as we don't use them, so we don't throw an error if they are missing
-  if (typeof session.id !== "string" || typeof session.exp !== "number") {
+
+  /**
+   * (EMB-829) Temporarily allow `exp` to be null or undefined while we're deprecating token without it
+   * after we disallow token without expiration, we will re-add this check.
+   */
+  if (typeof session.id !== "string") {
     throw MetabaseError.INVALID_SESSION_SCHEMA({
       expected: "{ id: string, exp: number, iat: number }",
       actual: JSON.stringify(session, null, 2),
     });
   }
 }
+
+export function isSessionTokenValid(session: any) {
```

Comment:
Used in `enterprise/frontend/src/embedding-sdk-bundle/store/reducer.ts`. I think this is cleaner than calling `validateSessionToken` there directly, since we don't need to wrap try/catch there, and the API remains the same by expecting a boolean.